### PR TITLE
Expose goodwe grid_export switch

### PR DESCRIPTION
**Pull request by zlymeda**
This change exposes the `grid_export` setting in addition to the `grid_export_limit`.
`grid_export` enables/disables the restriction of the export, while `grid_export_limit` defines how much can be exported.
When grid_export is enabled, the inverter takes the electricity from the grid even if we have enough PV to cover the house consumption (even if the export limit is high). It is beneficial to have grid_export off, and turn it on only in two scenarios:

- PV power is getting close to the reserved output
- electricity price is below 0 and it would cost money to sell the electricity

### CODE REVIEW

File: custom_components/goodwe/switch.py [hunk 1]
```diff
@@ -18,72 +18,120 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-LOAD_CONTROL = SwitchEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class GoodweSettingsSwitchEntityDescription(SwitchEntityDescription):
+    """Required values when describing Goodwe switch settings."""
+
+    icon_on: str
```
Comment (kukulich) [inline custom_components/goodwe/switch.py:26]: This should by done by icons translations see: https://developers.home-assistant.io/blog/2024/01/19/icon-translations

File: custom_components/goodwe/switch.py [hunk 2]
```diff
@@ -18,72 +18,120 @@
...
     entity_category=EntityCategory.CONFIG,
     device_class=SwitchDeviceClass.OUTLET,
+    settings_name='load_control_switch',
+)
+
+EXPORT_LIMIT_CONTROL = GoodweSettingsSwitchEntityDescription(
+    key="grid_export",
+    name="Grid Export",
+    icon_on="mdi:transmission-tower-off",
+    icon_off="mdi:transmission-tower-import",
```
Comment (kukulich) [inline custom_components/goodwe/switch.py:45]: Would not be better `mdi:transmission-tower-export` for on and `mdi:transmission-tower-off` for off?

File: custom_components/goodwe/switch.py [hunk 3]
```diff
@@ -18,72 +18,120 @@
...
+    icon_off="mdi:electric-switch",
+    icon_on="mdi:electric-switch-closed",
     entity_category=EntityCategory.CONFIG,
     device_class=SwitchDeviceClass.OUTLET,
+    settings_name='load_control_switch',
+)
+
+EXPORT_LIMIT_CONTROL = GoodweSettingsSwitchEntityDescription(
+    key="grid_export",
+    name="Grid Export",
```
Comment (kukulich) [inline custom_components/goodwe/switch.py:43]: Could you please change it to `translation_key` and set the name by translations? See https://github.com/mletenay/home-assistant-goodwe-inverter/pull/268

File: custom_components/goodwe/switch.py (hunk 1 above)
Comment (zlymeda) [reply custom_components/goodwe/switch.py:26]: it seems this is also a way: https://developers.home-assistant.io/docs/core/entity/#icon-property
I find it weird to extract this to a json and make things more complicated.

File: custom_components/goodwe/switch.py (hunk 3 above)
Comment (zlymeda) [reply custom_components/goodwe/switch.py:43]: will do

File: custom_components/goodwe/switch.py (hunk 1 above)
Comment (kukulich) [reply custom_components/goodwe/switch.py:26]: Json is the preferred way in HA now.

Comment (zlymeda) [reply custom_components/goodwe/switch.py:26]: I can either remove this and keep a single icon (which will be in the code, like all other entities currently in this repo) or keep it like this; or if you are willing to push the change to the PR, that'd be nice

Comment (kukulich) [reply custom_components/goodwe/switch.py:26]: Yes, I can prepare another commit for you.

Comment (okikaHA) [reply custom_components/goodwe/switch.py:26]: Hello, what is the status of this nice development ? thx

**Comment by mletenay** [comment]
There was no activity in past week and it got into conflict with upstream changes anyway, so I took the liberty and implemented this directly.
It was being requested for a long time ago, I just was somehow reluctant to do it when service call do the same.
Anyway, thanks for the suggestion/code.

**Pull request closed without merging**